elasticdl/pkg/ps/server_test.go

```diff
@@ -22,61 +27,171 @@ func createClient() (pb.PserverClient, context.Context, *grpc.ClientConn, contex
 	return c, ctx, conn, cancel
 }
 
-func TestMain(m *testing.M) {
+func TestPushModel(t *testing.T) {
 	// Create a PS server
 	serverDone := make(chan bool)
-	CreateServer(ADDR, serverDone)
-
-	result := m.Run()
-
-	os.Exit(result)
-}
-
-func TestPullVariable(t *testing.T) {
+	s, gs := CreateServer(ADDR, 0, "SGD", 0.1, serverDone)
 	client, ctx, conn, cancel := createClient()
 	defer conn.Close()
 	defer cancel()
-	request := pb.PullVariableRequest{}
-	_, err := client.PullVariable(ctx, &request)
+
+	var request pb.Model
+	// non embedding param
+	a := make([]float32, 10)
+	b := make([]float32, 10)
+	for i := 0; i < 10; i++ {
+		a[i] = rand.Float32()
+		b[i] = rand.Float32()
+	}
+	d := []int64{2, 5}
+	t1 := common.Tensor{"t1", a, d, nil}
+	t2 := common.Tensor{"t2", b, d, nil}
+
+	request.Param = append(request.Param, common.SerializeTensor(&t1))
+	request.Param = append(request.Param, common.SerializeTensor(&t2))
+
+	_, err := client.PushModel(ctx, &request)
+
 	if err != nil {
-		t.Errorf("Failed to pull variable")
+		t.Errorf("Failed to push model")
 	}
+
+	assert.True(t, s.Param.InitStatus)
+	assert.Len(t, s.Param.NonEmbeddingParam, 2)
+	assert.Contains(t, s.Param.NonEmbeddingParam, "t1")
+	assert.Contains(t, s.Param.NonEmbeddingParam, "t2")
+	assert.True(t, common.CompareFloatArray(a, s.Param.GetNonEmbeddingParam("t1").Value, 0.001))
+	assert.True(t, common.CompareFloatArray(b, s.Param.GetNonEmbeddingParam("t2").Value, 0.001))
```

Comment:
Do we need to check parameter version here?

Reply:
The model version in PS is directly set to the version in request. 

The key point of PushModel rpc is actually `InitStatus`. Several workers may push model to one PS, and once the model from one worker is  accepted, the others will be skipped.

We  have check the `InitStatus` already in this unit test.